---
title: "spatial-transcriptomics"
knit: (function(input_file, encoding) {
    out_dir <- 'docs';
    rmarkdown::render(input_file,
      encoding=encoding,
      output_file=file.path(dirname(input_file), out_dir, 'spatial-transcriptomics-report.html'))})
author: "Ronald Wu"
date: "2023-05-09"
output: html_document      
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Install Packages
```{r Install Packages}
# install.packages("devtools")
# 
# if (!base::requireNamespace("BiocManager", quietly = TRUE)){
#       install.packages("BiocManager")
#   }
# 
# BiocManager::install(c('BiocGenerics', 'DelayedArray', 'DelayedMatrixStats',
#                        'limma', 'S4Vectors', 'SingleCellExperiment',
#                        'SummarizedExperiment', 'batchelor', 'Matrix.utils', 'EBImage'))
# 
# install.packages("Seurat")
# 
# devtools::install_github(repo = "kueckelj/confuns")
# devtools::install_github(repo = "theMILOlab/SPATA2")
# 
# # if you want to use monocle3 related wrappers 
# devtools::install_github('cole-trapnell-lab/leidenbase')
# devtools::install_github('cole-trapnell-lab/monocle3')
# 
# # if you want to download data from MILOLab
# devtools::install_github("theMILOlab/SPATAData")
```

# MILOLab Database

```{r explore MILOLab database}

library(SPATAData)
library(SPATA2)

# overview of available datasets:
#View(source_df)
source_df

# load Shiny interactive downloader
# launchSpataData()

```

# MILOLab Database: Download SPATA objects
```{r download SPATA Objects from MILOLab database}

# Downloads single spata objects, saves them on your device and immediately
# returns them

#object <- downloadSpataObject(sample_name = "275_T")

# Downloads multiple spata objects at the same time  

#downloadSpataObjects(sample_names = c("275_T", "334_T"), folder = "objects")

```

# Visualization of GBM sample 275_T with SPATA2
```{r SPATA2 on GBM sample}
# load packages
library(SPATA2)

# ggplot2 plots can be easily combined with 'patchwork'
library(patchwork) 

# load objects
#human_cortex <- loadSpataObject("data/spata-obj-cortex265.RDS")
human_gbm <- loadSpataObject("objects/275_T.RDS")

# store example genes of interest as character vectors
genes_a <- c("TUBA1B", "HOPX", "PLP1", "ACTB")
genes_b <- c("CARTPT", "OLIG1", "GFAP", "SYNPR", "HOPX", "CCK")
```

```{r Basic Surface Plots}
# View FeatureNames
getFeatureNames(human_gbm)

# plot gene expression 
p1 <- 
  plotSurface(object = human_gbm, 
              color_by = "CA9",
              pt_size = 1.2,
              pt_clrsp = "magma"
              )

# plot gene expression (spatially smoothed)
p2 <- 
  plotSurface(object = human_gbm, 
              color_by = "CA9",
              pt_size = 1.2,
              pt_clrsp = "magma",
              smooth = TRUE, 
              smooth_span = 0.2)

# combine with patchwork 
p1 + legendNone() +
p2 + legendTop()
```

# View Sample Interactively
```{r interactive viewer}
# plots <- plotSurfaceInteractive(object = human_gbm)

```

# Next