<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ronald Wu" />

<meta name="date" content="2023-08-18" />

<title>01_interoperability</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="spatial-transcriptomics-report.html"></a>
    </li>
    <li>
      <a href="01_interoperability.html">01_interoperability</a>
    </li>
  </ul>
</li>
<li>
  <a href="speed-and-distance.html">speed-and-distance</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Whatever Menu
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="whatevermenu1.html">Whatever Menu 1</a>
    </li>
    <li>
      <a href="whatevermenu2.html">Whatever Menu 2</a>
    </li>
    <li>
      <a href="whatevermenu3.html">Whatever Menu 3</a>
    </li>
  </ul>
</li>
<li>
  <a href="unicorns.html">Unicorns</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">01_interoperability</h1>
<h4 class="author">Ronald Wu</h4>
<h4 class="date">2023-08-18</h4>

</div>


<div id="library-installation" class="section level1">
<h1>Library installation</h1>
<p>There was a major update to SPATA2 from version 0.1.0. Let’s upgrade
it to the current version 2.0.4.</p>
<p>We will also download the MILOLab spatial transcriptomic database
package <code>SPATAData</code> to help with ST dataset download.
Upgrading version from 0.0.0.9000 to new version:</p>
<pre class="r"><code>install.packages(&quot;devtools&quot;)

if (!base::requireNamespace(&quot;BiocManager&quot;, quietly = TRUE)){
      install.packages(&quot;BiocManager&quot;)
  }

BiocManager::install(c(&#39;BiocGenerics&#39;, &#39;DelayedArray&#39;, &#39;DelayedMatrixStats&#39;,
                       &#39;limma&#39;, &#39;S4Vectors&#39;, &#39;SingleCellExperiment&#39;,
                       &#39;SummarizedExperiment&#39;, &#39;batchelor&#39;, &#39;Matrix.utils&#39;, &#39;EBImage&#39;))

install.packages(&quot;Seurat&quot;)

devtools::install_github(repo = &quot;kueckelj/confuns&quot;)
devtools::install_github(repo = &quot;theMILOlab/SPATAData&quot;)
devtools::install_github(repo = &quot;theMILOlab/SPATA2&quot;)

# if you want to use monocle3 related wrappers 
devtools::install_github(&#39;cole-trapnell-lab/leidenbase&#39;)
devtools::install_github(&#39;cole-trapnell-lab/monocle3&#39;)</code></pre>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
<pre class="r"><code>library(SPATA2)</code></pre>
<pre><code>## The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,
## which was just loaded, will retire in October 2023.
## Please refer to R-spatial evolution reports for details, especially
## https://r-spatial.org/r/2023/05/15/evolution4.html.
## It may be desirable to make the sf package available;
## package maintainers should consider adding sf to Suggests:.
## The sp package is now running under evolution status 2
##      (status 2 uses the sf package in place of rgdal)</code></pre>
<pre class="r"><code>library(SPATAData)

library(Seurat)</code></pre>
<pre><code>## Attaching SeuratObject</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.2     ✔ readr     2.1.4
## ✔ forcats   1.0.0     ✔ stringr   1.5.0
## ✔ ggplot2   3.4.3     ✔ tibble    3.2.1
## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
## ✔ purrr     1.0.2</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="download-datasets" class="section level1">
<h1>Download datasets</h1>
<div id="x-genomics-in-house-gbm-visium-sample" class="section level2">
<h2>1. 10x Genomics in-house GBM Visium sample</h2>
<p>10X genomics houses a sample GBM Visium dataset <a
href="https://www.10xgenomics.com/resources/datasets/human-glioblastoma-whole-transcriptome-analysis-1-standard-1-2-0">here</a>.</p>
<p>We will download this dataset direct from the 10x genomics
website:</p>
<pre class="bash"><code>cd &quot;/Users/wuronald/WKLab Dropbox/Ronald Wu/_wouters lab/Data Ronald/spatial-transcriptomics/data&quot;
mkdir -p data/10xgenomics/GBM/</code></pre>
<pre class="bash"><code># Input Files

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_image.tif -P /data/10xgenomics/GBM

# Output Files

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_molecule_info.h5 -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_filtered_feature_bc_matrix.h5 -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_filtered_feature_bc_matrix.tar.gz -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_raw_feature_bc_matrix.h5 -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_raw_feature_bc_matrix.tar.gz -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_analysis.tar.gz -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_spatial.tar.gz -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_metrics_summary.csv -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_web_summary.html -P /data/10xgenomics/GBM

wget https://cf.10xgenomics.com/samples/spatial-exp/1.2.0/Parent_Visium_Human_Glioblastoma/Parent_Visium_Human_Glioblastoma_cloupe.cloupe -P /data/10xgenomics/GBM
</code></pre>
</div>
<div id="ravi-et-al-visium-gbm-sample-275_t" class="section level2">
<h2>2. Ravi et al Visium GBM sample (275_T):</h2>
<p>Using the <code>SPATAData</code> library, we will download one GBM
patient sample with the ID 275_T. There are two options to load the
275_T GBM ST sample:</p>
<ul>
<li><p>First, is to download the SPATA2 object directly:
<code>275_T.RDS</code>. This is done through the function
SPATAData::downloadSpataObjects(), which saves the object locally after
downloading it from dropbox. I have not gotten this to work due to a
<code>cannot read from connection error</code>.</p></li>
<li><p>Second, we can actually download the raw files.</p></li>
</ul>
<pre class="r"><code># download the object from dropbox
downloadSpataObjects(sample_names = c(&quot;275_T&quot;), folder = &quot;objects&quot;)
downloadSpataObjects(sample_names = c(&quot;313_T&quot;), folder = &quot;objects&quot;)

# download the object interactively via shinyapp
launchSpataData()</code></pre>
<p>Alternatively, we can use the dropbox link provided as part of the
error message and download it via a web browser.</p>
<pre class="r"><code># Meta data and links for MILO Lab database
SPATAData::source_df</code></pre>
<pre><code>## # A tibble: 34 × 21
##    sample patient_id age   stage status organ anatomical_region pathology
##    &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt; &lt;int&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;             &lt;chr&gt;    
##  1 275_T  #275UKF    &gt;=80      1 p      Brain frontal           Tumor    
##  2 270_T  #270UKF    40-50     1 p      Brain frontal           Tumor    
##  3 269_T  #269UKF    50-60     1 p      Brain occipital         Tumor    
##  4 268_T  #268UKF    40-50     1 p      Brain frontal           Tumor    
##  5 266_T  #266UKF    60-80     1 p      Brain temporal          Tumor    
##  6 265_C  #265UKF    50-60     1 h      Brain temporal          #N/A     
##  7 265_T  #265UKF    50-60     1 p      Brain temporal          Tumor    
##  8 262_T  #262UKF    50-60     1 p      Brain temporal          Tumor    
##  9 260_T  #260UKF    &gt;=80      1 p      Brain temporal          Tumor    
## 10 259_C  #259UKF    60-80     1 h      Brain temporal          #N/A     
## # ℹ 24 more rows
## # ℹ 13 more variables: hist_classification &lt;chr&gt;, hist_abbreviation &lt;chr&gt;,
## #   tags &lt;chr&gt;, organization &lt;chr&gt;, who_grade &lt;chr&gt;, species &lt;chr&gt;,
## #   link_spata &lt;chr&gt;, link_raw &lt;chr&gt;, link_image &lt;chr&gt;, citation &lt;chr&gt;,
## #   link_pub &lt;chr&gt;, date_added &lt;chr&gt;, assay_type &lt;chr&gt;</code></pre>
<pre class="r"><code># specify patient 275_T:
SPATAData::source_df %&gt;%
  filter(sample == &quot;275_T&quot;)</code></pre>
<pre><code>## # A tibble: 1 × 21
##   sample patient_id age   stage status organ anatomical_region pathology
##   &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt; &lt;int&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;             &lt;chr&gt;    
## 1 275_T  #275UKF    &gt;=80      1 p      Brain frontal           Tumor    
## # ℹ 13 more variables: hist_classification &lt;chr&gt;, hist_abbreviation &lt;chr&gt;,
## #   tags &lt;chr&gt;, organization &lt;chr&gt;, who_grade &lt;chr&gt;, species &lt;chr&gt;,
## #   link_spata &lt;chr&gt;, link_raw &lt;chr&gt;, link_image &lt;chr&gt;, citation &lt;chr&gt;,
## #   link_pub &lt;chr&gt;, date_added &lt;chr&gt;, assay_type &lt;chr&gt;</code></pre>
<pre class="r"><code># dropbox object link:
dropbox_obj_link &lt;-SPATAData::source_df %&gt;%
                  filter(sample == &quot;275_T&quot;) %&gt;%
                  select(&quot;link_spata&quot;) %&gt;% 
                  pull
# dropbox raw files link:
dropbox_raw_link &lt;-SPATAData::source_df %&gt;%
                  filter(sample == &quot;275_T&quot;) %&gt;%
                  select(&quot;link_raw&quot;) %&gt;% 
                  pull</code></pre>
<p>The following are the dropbox links for the 275_T sample:</p>
<ol style="list-style-type: decimal">
<li><p><a
href="https://www.dropbox.com/scl/fo/l5hmcuihkgdpdjwqx8brp/h?dl=1&amp;rlkey=i7i0x1iopc4w606ccqlsq3nix">RAW
files</a></p></li>
<li><p><a
href="https://www.dropbox.com/s/o9v04egctkpizta/275_T.RDS?dl=1">SPATA2
object</a></p></li>
</ol>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
