<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ronald Wu" />

<meta name="date" content="2023-05-09" />

<title>spatial-transcriptomics</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="spatial-transcriptomics-report.html">spatial</a>
</li>
<li>
  <a href="speed-and-distance.html">speed-and-distance</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Whatever Menu
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="whatevermenu1.html">Whatever Menu 1</a>
    </li>
    <li>
      <a href="whatevermenu2.html">Whatever Menu 2</a>
    </li>
    <li>
      <a href="whatevermenu3.html">Whatever Menu 3</a>
    </li>
  </ul>
</li>
<li>
  <a href="unicorns.html">Unicorns</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">spatial-transcriptomics</h1>
<h4 class="author">Ronald Wu</h4>
<h4 class="date">2023-05-09</h4>

</div>


<div id="install-packages" class="section level2">
<h2>Install Packages</h2>
<pre class="r"><code># install.packages(&quot;devtools&quot;)
# 
# if (!base::requireNamespace(&quot;BiocManager&quot;, quietly = TRUE)){
#       install.packages(&quot;BiocManager&quot;)
#   }
# 
# BiocManager::install(c(&#39;BiocGenerics&#39;, &#39;DelayedArray&#39;, &#39;DelayedMatrixStats&#39;,
#                        &#39;limma&#39;, &#39;S4Vectors&#39;, &#39;SingleCellExperiment&#39;,
#                        &#39;SummarizedExperiment&#39;, &#39;batchelor&#39;, &#39;Matrix.utils&#39;, &#39;EBImage&#39;))
# 
# install.packages(&quot;Seurat&quot;)
# 
# devtools::install_github(repo = &quot;kueckelj/confuns&quot;)
# devtools::install_github(repo = &quot;theMILOlab/SPATA2&quot;)
# 
# # if you want to use monocle3 related wrappers 
# devtools::install_github(&#39;cole-trapnell-lab/leidenbase&#39;)
# devtools::install_github(&#39;cole-trapnell-lab/monocle3&#39;)
# 
# # if you want to download data from MILOLab
# devtools::install_github(&quot;theMILOlab/SPATAData&quot;)</code></pre>
</div>
<div id="milolab-database" class="section level1">
<h1>MILOLab Database</h1>
<pre class="r"><code>library(SPATAData)
library(SPATA2)

# overview of available datasets:
#View(source_df)
source_df</code></pre>
<pre><code>## # A tibble: 34 Ã— 21
##    sample patient_id age   stage status organ anatomical_region pathology hist_classification     hist_abbreviation tags 
##    &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt; &lt;int&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;             &lt;chr&gt;     &lt;chr&gt;                   &lt;chr&gt;             &lt;chr&gt;
##  1 275_T  #275UKF    &gt;=80      1 p      Brain frontal           Tumor     Glioblastoma            GBM               #N/A 
##  2 270_T  #270UKF    40-50     1 p      Brain frontal           Tumor     Astrocytoma IDH mutated AA                #N/A 
##  3 269_T  #269UKF    50-60     1 p      Brain occipital         Tumor     Glioblastoma            GBM               #N/A 
##  4 268_T  #268UKF    40-50     1 p      Brain frontal           Tumor     Oligodendroglioma       ODG               #N/A 
##  5 266_T  #266UKF    60-80     1 p      Brain temporal          Tumor     Gliosarcoma             GBM               #N/A 
##  6 265_C  #265UKF    50-60     1 h      Brain temporal          #N/A      Cortex                  Cortex            #N/A 
##  7 265_T  #265UKF    50-60     1 p      Brain temporal          Tumor     Giant cell glioblastoma GBM               #N/A 
##  8 262_T  #262UKF    50-60     1 p      Brain temporal          Tumor     Glioblastoma            GBM               #N/A 
##  9 260_T  #260UKF    &gt;=80      1 p      Brain temporal          Tumor     Glioblastoma            GBM               #N/A 
## 10 259_C  #259UKF    60-80     1 h      Brain temporal          #N/A      Cortex                  GBM               #N/A 
## # â„¹ 24 more rows
## # â„¹ 10 more variables: organization &lt;chr&gt;, who_grade &lt;chr&gt;, species &lt;chr&gt;, link_spata &lt;chr&gt;, link_raw &lt;chr&gt;,
## #   link_image &lt;chr&gt;, citation &lt;chr&gt;, link_pub &lt;chr&gt;, date_added &lt;chr&gt;, assay_type &lt;chr&gt;</code></pre>
<pre class="r"><code># load Shiny interactive downloader
# launchSpataData()</code></pre>
</div>
<div id="milolab-database-download-spata-objects"
class="section level1">
<h1>MILOLab Database: Download SPATA objects</h1>
<pre class="r"><code># Downloads single spata objects, saves them on your device and immediately
# returns them

#object &lt;- downloadSpataObject(sample_name = &quot;275_T&quot;)

# Downloads multiple spata objects at the same time  

#downloadSpataObjects(sample_names = c(&quot;275_T&quot;, &quot;334_T&quot;), folder = &quot;objects&quot;)</code></pre>
</div>
<div id="visualization-of-gbm-sample-275_t-with-spata2"
class="section level1">
<h1>Visualization of GBM sample 275_T with SPATA2</h1>
<pre class="r"><code># load packages
library(SPATA2)

# ggplot2 plots can be easily combined with &#39;patchwork&#39;
library(patchwork) </code></pre>
<pre><code>## Warning: package &#39;patchwork&#39; was built under R version 4.2.3</code></pre>
<pre class="r"><code># load objects
#human_cortex &lt;- loadSpataObject(&quot;data/spata-obj-cortex265.RDS&quot;)
human_gbm &lt;- loadSpataObject(&quot;objects/275_T.RDS&quot;)</code></pre>
<pre><code>## 23:01:15 Loading spata-object.</code></pre>
<pre><code>## 23:01:22 Done.</code></pre>
<pre><code>## 23:01:22 Updating spata2 object.</code></pre>
<pre><code>## 23:01:22 Adding default for argument  &#39;min_lfc&#39; = 0.</code></pre>
<pre><code>## 23:01:22 Adding default for argument  &#39;pt_size_fixed&#39; = TRUE.</code></pre>
<pre><code>## 23:01:22 Converting variable &#39;segmentation&#39; to factor.</code></pre>
<pre><code>## 23:01:22 Done.</code></pre>
<pre class="r"><code># store example genes of interest as character vectors
genes_a &lt;- c(&quot;TUBA1B&quot;, &quot;HOPX&quot;, &quot;PLP1&quot;, &quot;ACTB&quot;)
genes_b &lt;- c(&quot;CARTPT&quot;, &quot;OLIG1&quot;, &quot;GFAP&quot;, &quot;SYNPR&quot;, &quot;HOPX&quot;, &quot;CCK&quot;)</code></pre>
<pre class="r"><code># View FeatureNames
getFeatureNames(human_gbm)</code></pre>
<pre><code>##             factor            numeric            integer 
##     &quot;segmentation&quot;   &quot;nCount_Spatial&quot; &quot;nFeature_Spatial&quot;</code></pre>
<pre class="r"><code># plot gene expression 
p1 &lt;- 
  plotSurface(object = human_gbm, 
              color_by = &quot;CA9&quot;,
              pt_size = 1.2,
              pt_clrsp = &quot;magma&quot;
              )</code></pre>
<pre><code>## 23:01:22 Normalizing values.</code></pre>
<pre><code>## 23:01:22 Done.</code></pre>
<pre class="r"><code># plot gene expression (spatially smoothed)
p2 &lt;- 
  plotSurface(object = human_gbm, 
              color_by = &quot;CA9&quot;,
              pt_size = 1.2,
              pt_clrsp = &quot;magma&quot;,
              smooth = TRUE, 
              smooth_span = 0.2)</code></pre>
<pre><code>## 23:01:22 Normalizing values.
## 23:01:22 Done.</code></pre>
<pre class="r"><code># combine with patchwork 
p1 + legendNone() +
p2 + legendTop()</code></pre>
<p><img src="spatial-transcriptomics-report_files/figure-html/Basic%20Surface%20Plots-1.png" width="672" /></p>
</div>
<div id="view-sample-interactively" class="section level1">
<h1>View Sample Interactively</h1>
<pre class="r"><code># plots &lt;- plotSurfaceInteractive(object = human_gbm)</code></pre>
</div>
<div id="next" class="section level1">
<h1>Next</h1>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
